frontend/src/providers/ColormodeProvider.tsx

```diff
@@ -0,0 +1,57 @@
+import {
+  createContext,
+  useContext,
+  // useEffect,
+  useMemo,
+  // useState,
+  ReactNode,
+  useCallback,
+} from "react";
+
+interface ColorModeContextValue {
+  toggleTheme: () => void;
+}
+
+const ColorModeContext = createContext<ColorModeContextValue>(null as any);
+
+const ColorModeProvider = ({ children }: { children: ReactNode }) => {
+  // const [mode, setMode] = useState("light");
+
+  const toggleTheme = useCallback(() => {
+    // method 1
+    // setMode((prev) => (prev === "light" ? "dark" : "light"));
+
+    // method 2
+    if (!document.documentElement.classList.contains("dark")) {
+      document.documentElement.classList.add("dark");
+    } else {
+      document.documentElement.classList.remove("dark");
+    }
+  }, []);
+
+  // useEffect(() => {
+  //  method 1
+  //  if (mode === "light") {
+  //    document.documentElement.classList.add("dark");
+  //  } else {
+  //    document.documentElement.classList.remove("dark");
+  //  }
+  // }, [mode]);
```

Comment:
lets use method 1 so we can pass the theme state down to the toggle button. and remove the unnecessary comments.